gordo/client/io.py

```diff
@@ -16,15 +15,56 @@ class HttpUnprocessableEntity(BaseException):
     pass
 
 
-def _handle_response(resp: requests.Response) -> Union[dict, bytes]:
+class ResourceGone(BaseException):
+    """
+    Represents an error from an HTTP status code of 410: Gone.
+    Indicates that access to the target resource is no longer available at the origin
+    server and that this condition is likely to be permanent.
+
+    Used in our case for calling the server with a revision which is no longer used.
+    """
+
+    pass
+
+
+class BadGordoRequest(BaseException):
+    """
+    Represents a general 4xx bad request
+    """
+
+    pass
+
+
+class NotFound(BaseException):
+    """
+    Represents a 404
+    """
+
+    pass
+
+
+def _handle_response(
+    resp: requests.Response, resource_name: Optional[str] = None
+) -> Union[dict, bytes]:
     if 200 <= resp.status_code <= 299:
         is_json = resp.headers["content-type"] == "application/json"
         return resp.json() if is_json else resp.content
     else:
-        msg = f"Failed to get response: {resp.status_code}: {resp.content!r}"
+        if resource_name:
+            msg = (
+                f"We failed to get response while fetching resource: {resource_name}. "
+                f"Return code: {resp.status_code}. Return content: {resp.content!r}"
+            )
+        else:
+            msg = f"Failed to get response: {resp.status_code}: {resp.content!r}"
+
         if resp.status_code == 422:
-            raise HttpUnprocessableEntity()
+            raise HttpUnprocessableEntity(msg)
+        elif resp.status_code == 410:
+            raise ResourceGone(msg)
+        elif resp.status_code == 404:
+            raise NotFound(msg)
         elif 400 <= resp.status_code <= 499:
-            raise BadRequest(msg)
+            raise BadGordoRequest(msg)
```

Comment:
Would this be a suitable change?
```suggestion
resp.raise_for_status()
```
[raise_for_status docs](https://2.python-requests.org/en/v2.0-0/user/quickstart/#errors-and-exceptions)

Reply:
Yeah probably nicer, why re-invent the wheel lizm

Reply:
I changed my mind. It caused some lame changes upstream (where instead of just looking at the type of the exception we needed to look inside it to determine what path to take). I am still for it in the long run, just not in this PR.